src/js/lib/monster.ui.js

```diff
@@ -3215,21 +3215,119 @@ define(function(require) {
 			attributes: attributes
 		});
 	}
 
+	/**
+	 * Generates a key-value pair editor
+	 * @param  {jQuery} $target  Container to append the widget to.
+	 * @param  {Object} [options]  Editor options
+	 * @param  {Object} [options.data]  Key-value data, as a plain object
+	 * @param  {String} [options.inputName]  Input name prefix
+	 * @param  {Object} [options.i18n]  Custom label translations
+	 * @param  {Object} [options.i18n.addLink]  Add row link label
+	 * @param  {String} [options.i18n.keyPlaceholder]  Key input placeholder
+	 * @param  {String} [options.i18n.valuePlaceholder]  Value input placeholder
+	 */
+	function keyValueEditor($target, options) {
+		if (!($target instanceof $)) {
+			throw TypeError('"$target" is not a jQuery object');
+		}
+		var data = _.get(options, 'data', {});
+		var inputName = _.get(options, 'inputName', 'data');
+		var addLink = _.get(options, 'i18n.addLink');
+		var keyPlaceholder = _.get(options, 'i18n.keyPlaceholder');
+		var valuePlaceholder = _.get(options, 'i18n.valuePlaceholder');
+		if (!_.isPlainObject(data)) {
+			throw new TypeError('"options.data" is not a plain object');
+		}
+		if (!_.isNil(inputName) && !_.isString(inputName)) {
+			throw new TypeError('"options.inputName" is not a string');
+		}
+		if (!_.isNil(addLink) && !_.isString(addLink)) {
+			throw new TypeError('"options.i18n.addLink" is not a string');
+		}
+		if (!_.isNil(keyPlaceholder) && !_.isString(keyPlaceholder)) {
+			throw new TypeError('"options.i18n.keyPlaceholder" is not a string');
+		}
+		if (!_.isNil(valuePlaceholder) && !_.isString(valuePlaceholder)) {
+			throw new TypeError('"options.i18n.valuePlaceholder" is not a string');
+		}
+		var $editorTemplate = $(monster.template(monster.apps.core, 'monster-key-value-editor', {
+			addLink: addLink
+		}));
+		var $rowContainer = $editorTemplate.find('.monster-key-value-data-container');
+		var addRow = function(value, key, index) {
+			$rowContainer.append(monster.template(monster.apps.core, 'monster-key-value-editor-row', {
+				key: key,
+				value: value,
+				inputName: inputName + '[' + index + ']',
+				keyPlaceholder: keyPlaceholder,
+				valuePlaceholder: valuePlaceholder
+			}));
+		};
+		// Add initial rows
+		var counter = 0;
+		_.each(data, function(value, key, index) {
+			addRow(value, key, counter++);
+		});
+		// Bind events
+		$editorTemplate.find('.key-value-add').on('click', function(e) {
+			e.preventDefault();
+			addRow('', '', counter++);
+		});
+		$rowContainer.on('click', '.key-value-remove', function(e) {
+			e.preventDefault();
+			$(this).closest('.monster-key-value-editor-row').remove();
+		});
+		// Append editor
+		$target.append($editorTemplate);
+		return $editorTemplate;
+	}
+
+	/**
+	 * Merges HTML attributes, mapped as JSON objects
+	 * @param   {Object} object  Destination object
+	 * @param   {Object} source  Source object
+	 * @returns {Object}         Returns `object` after merge
+	 */
+	function mergeHtmlAttributes(object, source) {
```

Comment:
Not sure how this relates to the task at hand, it does not seem like this function is used by `monster.ui#keyValueEditor` in any way.

Reply:
To clarify, this does not need to be remove, just wondering why it was implemented alongside the feature since it is not related in any way.

Reply:
I added the function because initially I thought to also have a Handlebars helper, and wanted to allow the user to add more attributes to the HTML tag like it is done with the icon helpers. However, late in the development I dismissed the idea, because the editor was going to require some interactions via JQuery, and the Handlebars helpers only return plain text, not JQuery objects. I decided to leave the function there because it can be used for other Handlebars helpers in the future.